nservicebus/messaging/delayed-delivery.md

```diff
@@ -5,10 +5,21 @@ tags:
 - Defer
 ---
 
-When a message is sent it does not need to be dispatched immediately. You can specify a later time for the message to be delivered. Before version 6 this was achieved by using `bus.Defer(...)` instead of `bus.Send(...)`. In version 6, this is accomplished by calling methods on an instance of `SendOptions` that is passed into `Send(...)` calls.  
+When a message is sent it does not need to be dispatched immediately. it can be specified a later time for the message to be delivered. In Version 5 and below this can be achieved by using `bus.Defer(...)` instead of `bus.Send(...)`. In Version 6, this is accomplished by calling methods on an instance of `SendOptions` that is passed into `Send(...)` calls.  
 
 NOTE: Only send operations can be deferred. You cannot defer a publish or a reply. 
 
+## Difference between versions
+### Defer Handling
+In Versions 3-5 by calling `bus.Defer(...)` the handling of the message is deferred by certain amount of time. After that time the message is passed to be handled by the same endpoint, it means that endpoint need to have handler for deferred message to be able to handle it. Using this mechanism delayed delivery can be achieved by calling `bus.Defer(...)` and a having handler in which the actual `bus.Send(...)` will be called after elapsed amount of time. 
+
+This method can also be used to defer message handling on a receiver end of message processing. Calling this method in a message handler would cause this handler to be called again after given time.  
+
+Semantically `bus.Defer(...)` is equivalent to calling `bus.SendLocal(...)` after time condition is met. Read more about [sending local](/nservicebus/messaging/send-a-message.md#sending-to-self) 
+
+### Delay Delivery
+In Version 6 a slightly different approach was taken to extend `SendOptions` by adding `DelayDeliveryWith` method. This allows to defer the actual sending of a message to any endpoint. To achieve the behavior of delayed handling like in previous versions `RouteToThisEndpoint()` need to be used with `DelayDeliveryWith` to defer [send to self](/nservicebus/messaging/send-a-message.md#sending-to-self). 
```

Comment:
Snippet will be added. Regarding difference, there is one big difference.
v3-5 Defer - is always doing SendLocal. So endpoint that calls it will recieve the message after set of time
v6 DelayDeliveryWith - is used with Send. So you can send it to yourself (behavior the same as with Defer) or to a different endpoint (actually doing delayed delivery).

And this is the reason why I do this write up, to explain the difference.